### PR TITLE
add asciidoctor.js@1.5.5-4 w/ npm auto-update

### LINKED ISSUE #10129: [Request] Add asciidoctor-js
**Library name:** asciidoctor-js
**Git repository url:** https://github.com/asciidoctor/asciidoctor.js
**npm package url(optional):** https://www.npmjs.com/package/asciidoctor.js
**License(s):** MIT License
**Official homepage:** http://asciidoctor.org/
**Wanna say something? Leave message here:** This is very useful for all of the JavaScript developers of Asciidoctor. 

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #10129
Related issue(s): # #
@kennynaoh please help me review this PR, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/asciidoctor/asciidoctor.js
 * Official website (optional, not the repository): http://asciidoctor.org
 * NPM package url (optional): https://www.npmjs.com/package/asciidoctor.js
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 20
   - Count of stars: 163
   - Count of forks: 33
 * NPM download stats (optional):
   - Downloads in the last day: 187
   - Downloads in the last week: 3779
   - Downloads in the last month: 18117

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by Mogztter** [comment]
This looks good to me.
How did you generate the .map files ?

I probably need to do a file verification using a md5 checksum.

**Comment by PeterDaveHello** [comment]
@Mogztter we use [uglify-js](https://www.npmjs.com/package/uglify-js) to generate the map files.
https://github.com/PeterDaveHello/web-minify-helper/blob/f287dacdbc7b5674c7cb4e49cc6673bcedb404b4/minify.sh#L75-L78

File: ajax/libs/asciidoctor.js/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "license": "MIT",
+  "homepage": "http://asciidoctor.org/",
+  "filename": "asciidoctor.min.js",
+  "npmName": "asciidoctor.js",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/asciidoctor.+(js|css)"
```
Comment (PeterDaveHello) [inline ajax/libs/asciidoctor.js/package.json]: Why not just `"**/*"`?

**Comment by sufuf3** [comment]
@PeterDaveHello  I modified it.
@kennynaoh please help me review this PR, thank you.

**Comment by PeterDaveHello** [comment]
Looks like there are more versions on git repo, any reason not to use git auto-update but npm?

**Comment by Mogztter** [comment]
I think it's safer to use npm. We could potentially tag an "unstable"
version for testing purposes.

Versions published on npm are "stable".


Le 29 janv. 2017 6:54 PM, "Peter Dave Hello" <notifications@github.com> a
écrit :

Looks like there are more versions on git repo, any reason not to use git
auto-update but npm?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/cdnjs/cdnjs/pull/10455#issuecomment-275932157>, or mute
the thread

**Comment by PeterDaveHello** [comment]
@Mogztter thanks a lot

**Comment by PeterDaveHello** [comment]
@sufuf3 so what's your reason here? In this case, you'll need to manually add the missing versions from git repo, for those versions not on npm.

**Comment by sufuf3** [comment]
@PeterDaveHello  I manually add the missing versions from git repo.
@kennynaoh please help me review this PR, thank you.

**Comment by sufuf3** [comment]
@pvnr0082t I modified it, please help me review this PR, thank you.

**Comment by sufuf3** [comment]
@pvnr0082t I modified the commit description, please help me review this PR, thank you.

**Comment by PeterDaveHello** [comment]
In this case, as its author mentioned, I don't see a reason we should manually get those versions from git repo, @sufuf3 you should take a look at the discussion ...

**Comment by sufuf3** [comment]
@PeterDaveHello  so should it need to manually add the missing versions from git repo, for those versions not on npm?
Although, author said the npm are "stable", but maybe user may use the old versions from git repo, which not on npm.

**Comment by Mogztter** [comment]
The vast majority of the projects using Asciidoctor.js are using the latest version or at least version 1.5.5+
I think it will confuse users to have all the versions from git.

**Comment by PeterDaveHello** [comment]
@sufuf3 just use npm.

**Comment by sufuf3** [comment]
@PeterDaveHello I modified it, please help me review this PR. Thank you.

**Comment by Mogztter** [comment]
Thanks! 🎉

**Comment by PeterDaveHello** [comment]
@Mogztter you're welcome 😄

**Pull request merged**